### DISCUSSION TITLE
Connecting Big Query via MindsDB cloud

### DISCUSSION THREAD

**Discussion by saldanhad** [mindsdb/mindsdb#3469] [Get help] [answered]
The json service account file is stored in my system, however the Big Query handler for MindsDB cloud needs it in a url format, any idea how this can be done?

![image](https://user-images.githubusercontent.com/115748404/196262878-a7b3115f-3509-48d1-925d-a0c81eebf66d.png)

**Comment by ZoranPandovski** [comment, marked as answer]
Hi @saldanhad for MindsDB cloud you will need to upload the file on a filestorage service as s3 and reference the url to it as:

```sql
CREATE DATABASE bigquery_datasource
WITH ENGINE='bigquery',
PARAMETERS={
  "project_id": "badger-345908",
  "service_account_keys": {
    "url": "http://s3-us-east-1.amazonaws.com/bucket/"
  }
};
```

This is not the most secure way but we will add the auth option for the files soon.

**Reply by saldanhad** [reply]
Hi @ZoranPandovski thanks for clarifying!
